src/webgpu/api/validation/resource_usages/textureUsageInRender.spec.ts

```diff
@@ -818,3 +821,76 @@ g.test('unused_bindings_in_pipeline')
       encoder.finish();
     });
   });
+
+g.test('validation_scope')
+  .params(
+    poptions('scope', ['no-draw', 'same-draw', 'different-draws', 'different-passes'] as const)
```

Comment:
IMO this test would be much easier to read as 4 separate tests (`'validation_scope,no_draw`, etc.) with all of the common code in a helper function. It's hard to see exactly how the different `scope` values result in different function call orders.

Reply:
It looks like the code is too simple to construct a helper function. Because the common part is just a few lines to create a texture, two bind groups, a render pipeline, a command encoder, a render pass, and we need to return almost all of these objects because they are used later. BTW, can we return an array of different objects? 

I added some comments. It might help to understand the code.

Reply:
You can have the helper return `{ pipeline: GPURenderPipeline, bindGroup0: GPUBindGroup, bindGroup1: GPUBindGroup }` and destructure it with `const { pipeline, bindGroup0, bindGroup1 } = helper();`. Is that what you had in mind?

Reply:
Yeah. It works!